src/librustc_mir/transform/const_prop.rs

```diff
@@ -60,7 +60,7 @@ impl MirPass for ConstProp {
     }
 }
 
-type Const<'tcx> = (Value, ty::Ty<'tcx>, Span);
+type Const<'tcx> = (&'tcx Allocation, ty::Ty<'tcx>, Span);
 
```

Comment:
I think this needs to stay Value. Or where was it necessary to move to allocation?

Reply:
Yeah, I think const_prop should just create one Miri context, then it can create as many `Value`s as it likes and it will all be freed when it's done.